module/modmanager/manager_test.go

```diff
@@ -1433,127 +1435,150 @@ func TestModularDiscovery(t *testing.T) {
+			delete(expectedStderr, line)
+		}
+		test.That(t, expectedStderr, test.ShouldBeEmpty)
 
-	stderr = logs.FilterMessage("got stderr").FilterLevelExact(zapcore.WarnLevel)
-	test.That(t, stderr.Len(), test.ShouldEqual, 2)
-	expectedStderr = map[string]struct{}{
-		"hiccup 1": {},
-		"hiccup 2": {},
-	}
-	for _, msg := range stderr.All() {
-		line := msg.ContextMap()["output"].(string)
-		delete(expectedStderr, line)
-	}
-	test.That(t, expectedStderr, test.ShouldBeEmpty)
+		test.That(t, logs.FilterMessage("first run script succeeded").Len(), test.ShouldEqual, 1)
 
-	test.That(t, logs.FilterMessage("first run script succeeded").Len(), test.ShouldEqual, 1)
+		t.Log("=== FIRST RUN SKIPPED AFTER SUCCESS ===")
 
-	t.Log("=== FIRST RUN SKIPPED AFTER SUCCESS ===")
+		logs.TakeAll() // remove logs observed up to this point
 
-	logs.TakeAll() // remove logs observed up to this point
+		err = mgr.FirstRun(ctx, modCfg)
+		test.That(t, err, test.ShouldBeNil)
 
-	err = mgr.FirstRun(ctx, modCfg)
-	test.That(t, err, test.ShouldBeNil)
+		test.That(t, logs.FilterMessage("first run already ran").Len(), test.ShouldEqual, 1)
 
-	test.That(t, logs.FilterMessage("first run already ran").Len(), test.ShouldEqual, 1)
+		t.Log("FIRST RUN SKIPPED AFTER SUCCESS AND MODULE MANAGER RESTART")
 
-	t.Log("FIRST RUN SKIPPED AFTER SUCCESS AND MODULE MANAGER RESTART")
+		logs.TakeAll() // remove logs observed up to this point
 
-	logs.TakeAll() // remove logs observed up to this point
+		err = mgr.Close(context.Background())
+		test.That(t, err, test.ShouldBeNil)
 
-	err = mgr.Close(context.Background())
-	test.That(t, err, test.ShouldBeNil)
+		opts = modmanageroptions.Options{
+			UntrustedEnv: false,
+		}
+		mgr = setupModManager(t, ctx, parentAddr, logger, opts)
 
-	opts = modmanageroptions.Options{
-		UntrustedEnv: false,
-	}
-	mgr = setupModManager(t, ctx, parentAddr, logger, opts)
+		err = mgr.FirstRun(ctx, modCfg)
+		test.That(t, err, test.ShouldBeNil)
 
-	err = mgr.FirstRun(ctx, modCfg)
-	test.That(t, err, test.ShouldBeNil)
+		test.That(t, logs.FilterMessage("first run already ran").Len(), test.ShouldEqual, 1)
+	})
 
-	test.That(t, logs.FilterMessage("first run already ran").Len(), test.ShouldEqual, 1)
+	t.Run("with timeout", func(t *testing.T) {
+		ctx := context.Background()
+		logger := logging.NewTestLogger(t)
+		exePath := rtestutils.BuildTempModuleWithFirstRun(t, "module/testmodule")
+		modCfg := config.Module{
+			Name:            "test-module",
+			ExePath:         exePath,
+			FirstRunTimeout: 1 * time.Millisecond,
+		}
+		parentAddr := setupSocketWithRobot(t)
+		opts := modmanageroptions.Options{
+			UntrustedEnv: false,
+		}
+		mgr := setupModManager(t, ctx, parentAddr, logger, opts)
+		err := mgr.FirstRun(ctx, modCfg)
+		test.That(t, err, test.ShouldNotBeNil)
+
+		var errExit *exec.ExitError
+		test.That(t, errors.As(err, &errExit), test.ShouldBeTrue)
+		test.That(t, errExit.String(), test.ShouldContainSubstring, "signal: killed")
```

Comment:
I wonder how portable this assertion is

Reply:
Seems fine to me for now; thanks for comment. Doubt the assertion will hold on windows but it might on mac.